### PR TITLE
swarm bootnode, no p2p discovery

### PR DESCRIPTION
**Pull request by nonsense**
TODO:
- [x] set `nodiscover` to be the default option we start Swarm with
- [x] install 3 `swarm-bootnode`s in Azure/AWS and hardcode them, replacing the existing bootnodes.

NICE TO HAVE:
- [ ] disconnect bootnodes after we have a certain number of peers (known nodes?) - https://github.com/ethersphere/go-ethereum/issues/1149

### CODE REVIEW

File: swarm/network/kademlia.go [hunk 1]
```diff
@@ -283,6 +283,22 @@ func (k *Kademlia) SuggestPeer() (suggestedPeer *BzzAddr, saturationDepth int, c
 func (k *Kademlia) On(p *Peer) (uint8, bool) {
 	k.lock.Lock()
 	defer k.lock.Unlock()
+
```
Comment (zelig) [inline swarm/network/kademlia.go]: this wont work for pss only peers in future.
i recommend putting a check in here https://github.com/ethereum/go-ethereum/blob/master/swarm/network/hive.go#L158
if no pss or stream protocol running, do not call On and skip to return on line https://github.com/ethereum/go-ethereum/blob/master/swarm/network/hive.go#L171

**Review by zelig — COMMENTED** [review]
make sure this line https://github.com/ethereum/go-ethereum/blob/master/swarm/network/stream/delivery.go#L258 takes care of the bootnode not being requested

File: swarm/network/stream/delivery.go [hunk 2]
```diff
@@ -259,6 +259,11 @@ func (d *Delivery) RequestFromPeers(ctx context.Context, req *network.Request) (
 				//log.Warn("Delivery.RequestFromPeers: peer not found", "id", id)
 				return true
 			}
+			// nodes that do not provide stream protocol
```
Comment (zelig) [inline swarm/network/stream/delivery.go:262]: can this happen at all? i thought `getPeer` only returns stream peers

File: swarm/network/stream/stream.go [hunk 3]
```diff
@@ -924,3 +929,13 @@ func (api *API) SubscribeStream(peerId enode.ID, s Stream, history *Range, prior
 func (api *API) UnsubscribeStream(peerId enode.ID, s Stream) error {
 	return api.streamer.Unsubscribe(peerId, s)
 }
+
+// hasStreamCap check if p2p.Peer the "stream" capability.
+func hasStreamCap(p *p2p.Peer) (yes bool) {
```
Comment (zelig) [inline swarm/network/stream/stream.go]: Is it not better we put this function in `p2p/protocols` with an argument for protocol name?

File: swarm/network/stream/delivery.go (hunk 2 above)
Comment (janos) [reply swarm/network/stream/delivery.go:262]: >  make sure this line https://github.com/ethereum/go-ethereum/blob/master/swarm/network/stream/delivery.go#L258 takes care of the bootnode not being requested

Yes, that is correct, can you please add more context to this earlier request? Handling this case is actually not needed.

**Review by janos — APPROVED** [review]

**Comment by nonsense** [comment]
Closing as this is now opened upstream.

**Comment by nonsense** [comment]
Opened upstream.

**Pull request closed without merging**